donkeycar/pipeline/sequence.py

```diff
@@ -1,154 +1,113 @@
-
-from typing import (Any, Callable, Generic, Iterable, Iterator, List, Sized, Tuple, TypeVar)
+from typing import Any, Callable, Generic, Iterable, Iterator, List, Sized, \
+    Tuple, TypeVar, Union
 
 from donkeycar.pipeline.types import TubRecord
 
 # Note: Be careful when re-using `TypeVar`s.
 # If you are not-consistent mypy gets easily confused.
 
-R = TypeVar('R', covariant=True)
 X = TypeVar('X', covariant=True)
 Y = TypeVar('Y', covariant=True)
 XOut = TypeVar('XOut', covariant=True)
 YOut = TypeVar('YOut', covariant=True)
 
 
-class SizedIterator(Generic[X], Iterator[X], Sized):
+# This is a protocol type without explicitly using a `Protocol`
+# Using `Protocol` requires Python 3.7
+class SizedIterable(Generic[X], Iterable[X], Sized):
```

Comment:
Base class of the user objects, which are sized iterables - here `TubSequence` and `TfmShallowList`.